--- a/kernel/nv.c	2014-12-02 03:58:34.000000000 +0000
+++ b/kernel/nv.c	2015-03-21 10:28:08.062379864 +0000
@@ -2014,7 +2014,11 @@ long nv_kern_unlocked_ioctl(
     unsigned long i_arg
 )
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,19,0)
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
+#else
     return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+#endif
 }
 
 long nv_kern_compat_ioctl(
@@ -2023,7 +2027,11 @@ long nv_kern_compat_ioctl(
     unsigned long i_arg
 )
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,19,0)
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
+#else
     return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+#endif
 }
 
 /*
