--- a/kernel/nv-linux.h
+++ b/kernel/nv-linux.h
@@ -1028,7 +1028,11 @@ static inline int nv_execute_on_all_cpus
 #endif
 
 #if !defined(NV_VMWARE)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)
+#define NV_NUM_PHYSPAGES                get_num_physpages()
+#else
 #define NV_NUM_PHYSPAGES                num_physpages
+#endif
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
--- a/kernel/nv.c
+++ b/kernel/nv.c
@@ -16,8 +16,12 @@
 #include "rmil.h"
 
 #if defined(MODULE_LICENSE)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 12, 0)
+MODULE_LICENSE("GPL");
+#else
 MODULE_LICENSE("NVIDIA");
 #endif
+#endif
 #if defined(MODULE_INFO)
 MODULE_INFO(supported, "external");
 #endif
